frontend/lib/resolution/scope-queries.cpp

```diff
@@ -621,14 +807,29 @@ isScopeElseBlockOfConditionalWithIfVar(Context* context, const Scope* scope) {
 }
 
 // appends to result
+//
+// traceCurPath and traceResult support gathering additional information
+// for error messages. The expectation is that the common case is that
+// they are both nullptr.
+//
+// if traceCurPath is not nullptr, it will be updated to reflect the
```

Comment:
Unless I misunderstand, you only ever push to `traceCurPath` if both `traceCurPath` and `traceResult` are non-null.